### PR TITLE
Update `metadata-service` to latest version + docs

### PR DESCRIPTION
**Pull request by evantahler**

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/orchestrator/README.md [hunk 1]
```diff
@@ -23,125 +24,165 @@ Alternatively, you can use `pip` to install Poetry:
...
+([Github Action](https://github.com/airbytehq/airbyte/blob/master/.github/workflows/metadata_service_deploy_orchestrator_dagger.yml)).
+
+1. Update the version of your code (`../lib`) and update the version of the package in
+   `pyproject.toml`
+1. In this project (`../orchestrator`) Run `poetry lock --no-update` to bump the version of the
+   requirements you may have changed in
+   `airbyte-ci/connectors/metadata_service/orchestrator/poetry.lock`
+1. Push your changes to the `master` branch and the orchestrator will be automatically deployed to
+   Dagster Cloud.
+
```
Comment (evantahler) [inline airbyte-ci/connectors/metadata_service/orchestrator/README.md:148]: This was the path I took

File: airbyte-ci/connectors/metadata_service/orchestrator/poetry.lock [hunk 2]
```diff
@@ -1970,7 +1980,7 @@ files = [
 
 [[package]]
 name = "metadata-service"
-version = "0.3.3"
+version = "0.3.4"
```
Comment (evantahler) [inline airbyte-ci/connectors/metadata_service/orchestrator/poetry.lock:1983]: This is the important change

File: airbyte-ci/connectors/metadata_service/orchestrator/README.md (hunk 1 above)
Comment (evantahler) [reply airbyte-ci/connectors/metadata_service/orchestrator/README.md:148]: Note: There was no changelog to update?

**Review by alafanechere — APPROVED** [review]

**Review by erohmensing — APPROVED** [review]
LG, try it out!

On success, you should see an update in the `Updated` time [here](https://airbyte-connectors.dagster.cloud/prod/locations): 

image

(not sure what happened 3 days ago...)

**Comment by evantahler** [comment]
Deploying via https://github.com/airbytehq/airbyte/actions/runs/7963711268/job/21739843047

**Pull request merged**